### PR TITLE
[ONNX] remove outdated ImplicitCastType QA in onnx.rst

### PR DESCRIPTION
**Pull request by titaiwangms**
Extend work from: https://github.com/pytorch/pytorch/pull/80596
This PR removes outdated QA of ImplicitCastType , as the coverage is greatly increased with the introduction of onnx shape inference and scalar type analysis.

### CODE REVIEW

**Comment by justinchuby** [comment]
@BowenBao I realized

> Q: Does ONNX support implicit scalar datatype casting?
  No, but the exporter will try to handle that part. Scalars are exported as constant tensors.
  The exporter will try to figure out the right datatype for scalars. 

Is probability still valid. Is that accurate?

File: docs/source/onnx.rst [hunk 1]
```diff
@@ -567,31 +567,6 @@ Q: How to export models with primitive type inputs (e.g. int, float)?
   Support for primitive numeric type inputs was added in PyTorch 1.9.
   However, the exporter does not support models with str inputs.
```
Comment (justinchuby) [inline docs/source/onnx.rst:569]:
```suggestion

Q: Does ONNX support implicit scalar datatype casting?

  No, but the exporter will try to handle that part. Scalars are exported as constant tensors.
  The exporter will figure out the right data type for scalars. In rare cases when it is unable
  to do so, you will need to manually specify the datatype with e.g. `dtype=torch.float32`.
  If you see any errors, please create a GitHub issue.
```

**Review by justinchuby — CHANGES_REQUESTED** [review]
I would retain a part of the answer

**Review by justinchuby — APPROVED** [review]
LGTM🚀

**Review by BowenBao — APPROVED** [review]

**Pull request closed without merging**